uaclient/config.py

```diff
@@ -342,13 +342,19 @@ def entitlements(self):
             return {}
 
         self._entitlements = {}
-        contractInfo = machine_token["machineTokenInfo"]["contractInfo"]
+        contractInfo = machine_token.get("machineTokenInfo", {}).get(
+            "contractInfo"
+        )
+        if not contractInfo:
+            return {}
+
         tokens_by_name = dict(
-            (e["type"], e["token"])
+            (e.get("type"), e.get("token"))
             for e in machine_token.get("resourceTokens", [])
         )
         ent_by_name = dict(
-            (e["type"], e) for e in contractInfo["resourceEntitlements"]
+            (e.get("type"), e)
+            for e in contractInfo.get("resourceEntitlements", [])
```

Comment:
I think the default value here should be a dict, right ?

Reply:
No, `resourceEntitlements` is a list